internal/providers/unix/components/process/process.go

```diff
@@ -47,30 +48,24 @@ func (p *Process) start(ctx context.Context) error {
 		return errutil.WithHTTPStatus(http.StatusBadRequest, err)
 	}
 
-	childEnv, err := json.Marshal(p.Environment)
-	if err != nil {
-		return fmt.Errorf("encoding child environment")
-	}
-
 	// Construct supervised command.
 	supervisePath := os.Args[0]
-	superviseArgs := append(
-		[]string{
-			"supervise",
-			"--",
-			strconv.Itoa(int(p.SyslogPort)),
-			p.ComponentID,
-			p.WorkspaceRoot,
-			string(childEnv),
-			program,
-		},
-		p.Arguments...,
-	)
-	cmd := exec.Command(supervisePath, superviseArgs...)
+	cmd := exec.Command(supervisePath, "supervise")
 	cmd.SysProcAttr = &syscall.SysProcAttr{
 		Setsid: true, // Run in background.
 	}
 
+	// Pipe JSON config to supervise on stdin.
+	configJSON := supervise.MustEncodeConfig(&supervise.Config{
```

Comment:
I don't see this function defined anywhere...

Reply:
I just added the file that *should* have been there initially. Thanks for catching this!